optuna/visualization/_edf.py

```diff
@@ -0,0 +1,138 @@
+                b = math.exp(0.5 * (math.cos(2 * math.pi * x) + math.cos(2 * math.pi * y)))
+                return -a - b + math.e + 20
+
+
+            def objective(trial, low, high):
+                x = trial.suggest_float("x", low, high)
+                y = trial.suggest_float("y", low, high)
+                return ackley(x, y)
+
+
+            sampler = optuna.samplers.RandomSampler()
+
+            # Widest search space.
+            study0 = optuna.create_study(study_name="x=[0,5), y=[0,5)", sampler=sampler)
+            study0.optimize(lambda t: objective(t, 0, 5), n_trials=500)
+
+            # Narrower search space.
+            study1 = optuna.create_study(study_name="x=[0,4), y=[0,4)", sampler=sampler)
+            study1.optimize(lambda t: objective(t, 0, 4), n_trials=500)
+
+            # Narrowest search space but it doesn't include the global optimum point.
+            study2 = optuna.create_study(study_name="x=[1,3), y=[1,3)", sampler=sampler)
+            study2.optimize(lambda t: objective(t, 1, 3), n_trials=500)
+
+            optuna.visualization.plot_edf([study0, study1, study2])
+
+        .. raw:: html
+
+            <iframe src="../../_static/plot_edf.html"
+             width="100%" height="500px" frameborder="0">
+            </iframe>
+
+    Args:
+        study:
+            A target :class:`~optuna.study.Study` object.
+            You can pass multiple studies if you want to compare those EDFs.
+
+    Returns:
+        A :class:`plotly.graph_objs.Figure` object.
+    """
+
+    _imports.check()
+
+    if isinstance(study, list):
+        studies = study
+    else:
+        studies = [study]
+
+    return _get_edf_plot(studies)
+
+
+def _get_edf_plot(studies: List[Study]) -> "go.Figure":
+    if len(studies) == 0:
+        raise ValueError("No studies were given.")
+
+    layout = go.Layout(
+        title="EDF Plot",
+        xaxis={"title": "Objective Value"},
+        yaxis={"title": "Cumulative Probability"},
+    )
+
+    all_trials = list(
+        itertools.chain.from_iterable(
+            (
+                trial
+                for trial in study.get_trials(deepcopy=False)
+                if trial.state == TrialState.COMPLETE
+            )
+            for study in studies
+        )
+    )
+
+    if len(all_trials) == 0:
+        _logger.warning("There are no complete trials.")
+        return go.Figure(data=[], layout=layout)
+
+    min_x_value = min(trial.value for trial in all_trials)
+    max_x_value = max(trial.value for trial in all_trials)
+    x_values = np.linspace(min_x_value, max_x_value, 100)
+
+    traces = []
+    for study in studies:
+        trials = [
+            trial
+            for trial in study.get_trials(deepcopy=False)
+            if trial.state == TrialState.COMPLETE
+        ]
+
+        y_values = []
+        for x in x_values:
+            y = len([trial for trial in trials if trial.value <= x]) / len(trials)
+            y_values.append(y)
```

Comment:
Somewhat simplify-able.
```suggestion
        values = np.asarray(
            [
                trial.value
                for trial in study.get_trials(deepcopy=False)
                if trial.state == TrialState.COMPLETE
            ]
        )

        y_values = np.sum(values[:, np.newaxis] <= x_values, axis=0) / values.size
```

Reply:
Cool!